api/src/main/java/net/md_5/bungee/api/event/ResourcePackRequestEvent.java

```diff
@@ -0,0 +1,67 @@
+package net.md_5.bungee.api.event;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.connection.Connection;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.api.plugin.Cancellable;
+import net.md_5.bungee.api.plugin.Event;
+
+/**
+ * Called when a server sends a request for a resource pack to
+ * the client
+ */
+@Getter
+@RequiredArgsConstructor
+@ToString
+@EqualsAndHashCode(callSuper = false)
+public class ResourcePackRequestEvent extends Event implements Cancellable
+{
+
+    /**
+     * Cancelled state.
+     */
+    @Setter
+    private boolean cancelled;
+
+    /**
+     * The one who sent the request.
+     */
+    private final Connection sender;
```

Comment:
This would suggest a targeted event, but it raises problems that the API calls this.

Why do we even need the API to call an event? It's not particularly usual in plugins

Reply:
> Why do we even need the API to call an event?

the packet for some reason, whenever sent from bungee, does not get captured by neither Upstream/DownstreamBridge nor ServerConnector, hence the event not called. that's why this event is getting called when the method gets called. there are other methods in bungeee that do the same for other reasons.

Reply:
more information on the API calling the event: current implementation for sending packets does not call any handler whatsoever ; so how otherwise shall we call the event?